bindings/ruby/lib/rock_gazebo/syskit/robot_definition_extension.rb

```diff
@@ -359,12 +366,52 @@ def normalize_name(name)
                 name.gsub(/:+/, '_')
             end
 
+            # @api private
+            #
+            # Describes recursively all sensors and plugins in the model
+            def load_gazebo_robot_submodels(
+                sdf_model,
+                name: sdf_model.name, prefix_device_with_name: true,
+                deployment_prefix: "",
+                is_submodel: false
+            )
+
+                sdf_model.each_sensor do |sensor|
+                    gazebo_define_sensor_device(
+                        sdf_model, sensor,
+                        model_name: name,
+                        prefix_device_with_name: prefix_device_with_name,
+                        deployment_prefix: is_submodel ? deployment_prefix + ":" : deployment_prefix
+                    )
+                end
+
+                sdf_model.each_plugin do |plugin|
+                    gazebo_define_plugin_device(
+                        sdf_model, plugin,
+                        model_name: name,
+                        prefix_device_with_name: prefix_device_with_name,
+                        deployment_prefix: is_submodel ? deployment_prefix + ":" : deployment_prefix
+                    )
+                end
+
+                sdf_model.each_model do |sdf_submodel|
+                    load_gazebo_robot_submodels(
+                        sdf_submodel,
+                        name: name + "_" + sdf_submodel.name,
+                        prefix_device_with_name: true,
+                        deployment_prefix: deployment_prefix + ":" + sdf_submodel.name,
+                        is_submodel: true
```

Comment:
Don't like this is_submodel here. We should be able to do without by changing the name generation logic to handle both cases, just like we did for the orogen model side.